Classic client support

Vladinator opened the pull request:
These changes make it so the addon can load in the Classic client.

We might need to look at feature improvements or additions specific to Classic, but the framework code changes required are done and working.

Both retail and classic use the same code-base so it is easy to keep up-to-date.

If we ever need to branch logic we can check `IS_RETAIL` or `IS_CLASSIC` and act accordingly.

Vladinator committed:
First code-pass to get the basics ready for classic.

A lot of docs changed so had to also improve those to get rid of warnings.

jahraphael commented on core.lua:
```diff
@@ -1,4 +1,6 @@
-if WOW_PROJECT_ID ~= WOW_PROJECT_MAINLINE then return end
+local IS_RETAIL = WOW_PROJECT_ID == WOW_PROJECT_MAINLINE
+local IS_CLASSIC = WOW_PROJECT_ID == WOW_PROJECT_CLASSIC
```
There's one more variant of Classic we need to support: `WOW_PROJECT_WRATH_CLASSIC`

Maybe for now we could just make IS_CLASSIC return true if the project is `WOW_PROJECT_WRATH_CLASSIC` OR `WOW_PROJECT_CLASSIC`?

Vladinator replied:
I added a `IS_WRATH` flag for later use in the code. So we consider all 3 editions of the game.

jahraphael commented on core.lua:
```diff
@@ -189,6 +191,7 @@ do
     ns.LOOKUP_MAX_SIZE = floor(2^18-1) -- the maximum index we can use in a table before we start to get errors
     ns.CURRENT_SEASON = 1 -- the current mythic keystone season. dynamically assigned once keystone data is loaded.
     ns.RAIDERIO_ADDON_DOWNLOAD_URL = "https://rio.gg/addon"
+    ns.RAIDERIO_DOMAIN = IS_CLASSIC and "classic.raider.io" or "raider.io"
```
The domains will need to be modified to match the specific site:

WOW_PROJECT_MAINLINE => `raider.io`
WOW_PROJECT_WRATH_CLASSIC => `classic.raider.io`
WOW_PROJECT_CLASSIC => `era.raider.io`

Since we can expect there will be future versions of classic, we may want to be able to bundle the classic checks. So it's:

WOW_PROJECT_CLASSIC ==> era
WOW_PROJECT_WRATH_CLASSIC || WOW_PROJECT_BURNING_CRUSADE_CLASSIC ==> classic

Vladinator replied:
I expanded the line to select the appropriate domain based on the `IS_CLASSIC` and `IS_WRATH` checks.

jahraphael reviewed (changes requested):
A couple of little things to address due to the multiple versions of Classic.

Vladinator committed:
- Added Wrath TOC.

- Added Wrath check and using it to select the approriate RIO domain for `ns.RAIDERIO_DOMAIN`.
